packages/makeup-active-descendant/test/index.js

```diff
@@ -1,20 +1,373 @@
+        var focusEl = widgetEl.querySelector('input');
+
+        beforeEach(function() {
+            document.body.innerHTML = dom;
+            testActiveDescendant = ActiveDescendant.createLinear(widgetEl, focusEl, containerEl, 'li', { wrap: true }); // eslint-disable-line
+
+            onActiveDescendantChange = jasmine.createSpy('onActiveDescendantChange');
+            widgetEl.addEventListener('activeDescendantChange', onActiveDescendantChange);
+
+            focusEl.focus();
+        });
+
+        afterEach(function() {
+            testActiveDescendant.destroy();
+            onActiveDescendantChange.calls.reset();
+        });
+
+        it('should trigger onActiveDescendantChange event on arrow left', function() {
+            // execute
+            widgetEl.dispatchEvent(new CustomEvent('arrowLeftKeyDown', { detail: { target: { tagName: '' } } }));
+            // assert
+            expect(onActiveDescendantChange).toHaveBeenCalledTimes(1);
+            // execute
+            onActiveDescendantChange.calls.reset();
+            widgetEl.dispatchEvent(new CustomEvent('arrowLeftKeyDown', { detail: { target: { tagName: '' } } }));
+            // assert
+            expect(onActiveDescendantChange).toHaveBeenCalledTimes(1);
+        });
+
+        it('should trigger onActiveDescendantChange event on arrow up', function() {
+            // execute
+            widgetEl.dispatchEvent(new CustomEvent('arrowUpKeyDown', { detail: { target: { tagName: '' } } }));
+            // assert
+            expect(onActiveDescendantChange).toHaveBeenCalledTimes(1);
+            // execute
+            onActiveDescendantChange.calls.reset();
+            widgetEl.dispatchEvent(new CustomEvent('arrowUpKeyDown', { detail: { target: { tagName: '' } } }));
+            // assert
+            expect(onActiveDescendantChange).toHaveBeenCalledTimes(1);
+        });
+    });
+
+    describe('when module is created in programmatic relationship with autoInit', function() {
+        var focusEl = widgetEl.querySelector('input');
+
+        beforeEach(function() {
+            document.body.innerHTML = dom;
+            testActiveDescendant = ActiveDescendant.createLinear(widgetEl, focusEl, containerEl, 'li', { autoInit: 2 }); // eslint-disable-line
+
+            onActiveDescendantChange = jasmine.createSpy('onActiveDescendantChange');
+            widgetEl.addEventListener('activeDescendantChange', onActiveDescendantChange);
+        });
+
+        afterEach(function() {
+            testActiveDescendant.destroy();
+            onActiveDescendantChange.calls.reset();
+        });
+
+        it('should set aria-activedescendant to last element child', function() {
+            // eslint-disable-next-line max-len
+            expect(focusEl.getAttribute('aria-activedescendant')).toEqual(containerEl.lastElementChild.getAttribute('id'));
+        });
+    });
+
+    // describe('when module is created in programmatic relationship with autoReset', function() {
```

Comment:
Unable to set the focus to the input elements to set autoReset. focus remains at the document level.